### DISCUSSION TITLE
New Contributor Request

### DISCUSSION THREAD

**Discussion by NTCoding** [ddd-crew/ddd-starter-modelling-process#57] [Team Posts]
Hey Everyone,

I received this awesome email this morning from someone who would like to contribute back to the DDD Crew. They would like to share a questionnaire they use for assessing people's DDD level and using that to identify what training opportunities are needed. He's very keen to share and allow the community to evolve the content too.

It definitely seems like a valuable addition to the DDD Crew toolkit to me.

If I get a couple of thumbs up on this post I'll create an empty repo called "DDD Learning Assessment" and invite Alex to submit a PR.

> Good morning Nick, 
> 
> First of all, congratulations on all the fantastic content you created either in ddd-crew and books 👏
> 
> 
> I'm Aleix Morgadas, I have been studying DDD for several years, and ddd-crew is a remarkable resource to be inspired and learn.
> 
> 
> Lately, I'm assessing DDD company knowledge and creating the necessary material to run workshops to train team members on DDD. I have been thinking about contributing back to the community, and I thought about sharing a Google Form that we are using to Assess people's knowledge about DDD within an organization.
> 
> [This form](https://forms.gle/KCLuFz3BsFP4TJU16) is to understand people's knowledge and gaps and then focus the training on the specific instead of the general to get the most value.
> 
> Is there a way to contribute back to the community and publish those materials in ddd-crew under the CC License? Of course, that survey needs to be evolved, but that's part of the beauty of an open community 😄
> 
> Thank you very much for your time,
> 
> Aleix Morgadas​

**Comment by yellowbrickc** [comment]
You have 5 x 👍 from me. You already added a quiz, also a kind of questionnaire in a private repo. We could add this "capability" to the repos where it fits, additionally to the explanations, links, etc? It would surely help. I imagine that the questions would also transport the heuristics we use. Asking the right questions is not easy, a guide would be really helpful.

My next step would be to invite Aleix to the best fitting repo (I think either the https://github.com/ddd-crew/welcome-to-ddd or the https://github.com/ddd-crew/ddd-starter-modelling-process) so that he can start the discussion. We should not only discuss the questions but also the tooling (it would be cool if we could keep everything in versioned on github)

What do you think @ddd-crew/crew?

**Comment by emgsilva** [comment]
👍 from me too! This can be an interesting resource to "curate and improve together". (Maybe in time we can mention some ideas of interesting resources to go for as a function of where people are...).

I agree with @yellowbrickc: would be great to have a versioned form of the survey (so instead of Google from, have it as Markdown in one GitHub repo). This then can be implemented by people on Google Form if that makes sense, but I can see this being useful for other situations, for example when starting a workshop in person or remote (in Miro), or something like that.

I would say to keep the survey as a dedicated repo (like the name @NTCoding suggestions) - and we may consolidate other ideas there too. Would then be good to link from those starting repos, such as the one's @yellowbrickc mentions.

**Comment by Baasie** [comment]
It is a really nice and useful sense-making tool.

My worry is that this will end up as a maturity model, probably not the intention but we need to keep an eye on that :)

**Comment by heimeshoff** [comment]
agreed, this looks quite helpful in sense and decision making, and we have
all the power to not let it become a maturity model <3

**Comment by NTCoding** [comment]
Thanks for the good comments looking at this from different perspectives. Definitely agree with @Baasie that this is a good sense-making tool for teams to identify where they are and what their next steps might be etc.

Any opinions on the name of the repo? Should we try to imply sense-making in the name so that the intention is explicit and will help to avoid morphing into a maturity model?

**Comment by yellowbrickc** [comment]
Do you really want to put it in a separate repo? @emgsilva, you have proposed it, but you didn't say why?

IMO having a dedicated repo is where the danger begins. This could signal that it is just as important as the other repos. And this is not true; it is only an additional resource to verify the knowledge you already have and find the gaps you can fill with the help of **other** resources. 

(it seems that it's challenge-time 😁, sorry)

**Comment by emgsilva** [comment]
Great point @yellowbrickc (sharp as always 😄)   

I was following/building on Nick's original suggestion (`I'll create an empty repo called "DDD Learning Assessment"`). But to be honest this could go both ways:

- **keep it as a separate repo** ("ddd-{knowledge, awareness, ??}-sense-making"??) - serve the purpose of sensing where teams are in terms of DDD knowledge and awareness - so they can best proceed on their "DDD journey". I think that makes it a good "capability" to have on our "library" of tools, which can be used in different activities and moments. 
- **make it part of journey-repos** - could just be integrated into one of those repos (as a file or folder in the repo). This could help us be more explicit about usage and not open up space for using this in other situations.

My impression is that we have been adopting the principle of separating each thing in its own repo, and having a few "orchestrating/user-journey" repos that we link and provide some "default journeys" to approach DDD. So I would probably go for that option. But I could see making this part of one of the "user-journey repos" to help minimize the concern of misusing this as a "maturity model"... but isn't that a bit too restrictive?

Either way, I propose we align a bit and keep the ball rolling and adapt if get signals and learn that things can be improved.

**Comment by yellowbrickc** [comment]
Thanks, Eduardo, I give my best 😄 

I understand now the reasons for this proposal, and I fully agree: keep the ball rolling. I like the names you've proposed, especially **ddd-awareness**. (ddd-knowledge would be a little bit too much. Everything here is ddd-knowledge)

One last sentence to explain my thoughts: my concern was not so much the threat of a maturity model (which is a genuine concern, of course) but the danger of offering munition for any certifications 😱 

So @NTCoding GO-GO-GO! Pick one name and invite Aleix :) (None of these decisions is irreversible, we can change names, reorganize repos later if we need)

**Comment by NTCoding** [comment]
Thanks @yellowbrickc. You're right that we can change the name later so it's not super important to get it right up-front.

Here's what Alex suggested:

> ddd-familiarity-survey <- this one makes more sense to me
> ddd-familiarity-assessment <- this follows the trend of https://github.com/TeamTopologies/Team-Cognitive-Load-Assessment, it might make sense
> I add the familiarity to ensure it's not about being right or wrong, but to know which areas people are more familiar with and help them feel that they don't need to know everything to answer the survey.

I did a quick random select of all the options mentioned and it picked "ddd-familiarity-assessment" so I'll go with that for now!
